GBM model type (#173)
ludwig-ai/ludwig-docs: docs/configuration/trainer.md

    - `optimizer` (default `{type: adam, beta1: 0.9, beta2: 0.999, epsilon: 1e-08}`): which optimizer to use with the relative parameters. The available optimizers are: `sgd` (or `stochastic_gradient_descent`, `gd`, `gradient_descent`, they are all the same), `adam`, `adadelta`, `adagrad`, `adamax`, `ftrl`, `nadam`, `rmsprop`. Check [PyTorch optimizer documentation](https://pytorch.org/docs/stable/optim.html) for a full list of parameters for each optimizer. The optimizer definition can also specify gradient clipping using `clipglobalnorm`, `clipnorm`, and `clipvalue`.

=== "GBM"

    See the [LightGBM documentation](https://lightgbm.readthedocs.io/en/latest/Parameters.html) for more details about the available parameters.

    - `type` (default `lightgbm_trainer`): Trainer to use for training the model. Must be one of ['lightgbm_trainer'] - corresponds to name in `ludwig.trainers.registry.(ray_)trainers_registry`.

>     - `boosting_type` (default `gbdt`): Type of boosting algorithm to use.

Reviewer: If we do also support [random forest](https://lightgbm.readthedocs.io/en/latest/Parameters.html#boosting) and the two other boosting types, I think it would be really good to mention this here since this is an important top-level parameter.

    - `bagging_freq` (default `0`): Frequency of bagging.
    - `pos_bagging_fraction` (default `1.0`): Fraction of positive data to use for bagging.
    - `feature_fraction` (default `1.0`): Fraction of features to use.
    - `eval_batch_size` (default `128`): Size of batch to pass to the model for evaluation.
    - `bagging_fraction` (default `1.0`): Fraction of data to use for bagging.
    - `uniform_drop` (default `False`): Whether to use uniform dropout. Used only with boosting_type 'dart'.
    - `validation_field` (default `combined`): First output feature, by default it is set as the same field of the first output feature.
    - `max_drop` (default `50`): Maximum number of dropped trees during one boosting iteration. Used only with boosting_type 'dart'. A negative value means no limit.
    - `verbose` (default `0`): Verbosity level for GBM trainer.

## Optimizer parameters

=== "ECD"

    The available optimizers wrap the ones available in PyTorch.
    For details about the parameters that can be used to configure different optimizers, please refer to the [PyTorch documentation](https://pytorch.org/docs/stable/optim.html).

>     The `learning_rate` parameter the optimizer will use come from the `trainer` section.

Reviewer: ```suggestion
    The `learning_rate` parameter used by the optimizer comes from the `trainer` section.
```
